### DISCUSSION TITLE
How to start JanusGraph with debug?

### DISCUSSION THREAD

**Discussion by qDonl** [JanusGraph/janusgraph#2655] [Q&A]
I have a program embedded in JanusGraph. This program was started by JanusGraph,  I want to remotely debug through IDEA, but I don’t know how to start JanusGraph in debug mode.
I want to start JanusGraph like a JAR.

```bash
java -Xdebug -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=9999 -jar myjar.jar
```
but i don;t know how to start janusgraph with debug.

Please show detailed commands, thx!!!

**Comment by vtslab** [comment]
It seems you follow the tutorial: [https://www.jetbrains.com/help/idea/tutorial-remote-debug.html](https://www.jetbrains.com/help/idea/tutorial-remote-debug.html)
This tutorial already gives detailed instructions how to start your application jar. When trying this, what happens?
Note that your application jar uses embedded janusgraph, so janusgraph with its dependencies should be on the classpath. Do you use maven to specify dependencies for your application?
